// react-spotify-web-playback/esm/components/Actions.js
import * as React from 'react';
import { px, styled } from '../styles';
import Devices from './Devices';
import Volume from './Volume';
var Wrapper = styled('div')({
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'flex-end',
    paddingRight: px(10),
    'pointer-events': 'none',
    '> div + div': {
        marginLeft: px(10),
    },
    '@media (max-width: 1023px)': {
        bottom: 0,
        position: 'absolute',
        right: 0,
        width: 'auto',
    },
}, function (_a) {
    var styles = _a.styles;
    return ({
        height: px(styles.height),
    });
}, 'ActionsRSWP');
function Actions(props) {
    var currentDeviceId = props.currentDeviceId, devices = props.devices, isDevicesOpen = props.isDevicesOpen, onClickDevice = props.onClickDevice, setVolume = props.setVolume, styles = props.styles, volume = props.volume;
    return (React.createElement(Wrapper, { styles: styles },
        currentDeviceId && React.createElement(Volume, { volume: volume, setVolume: setVolume, styles: styles }),
        React.createElement(Devices, { deviceId: currentDeviceId, devices: devices, open: isDevicesOpen, onClickDevice: onClickDevice, styles: styles })));
}
export default Actions;
