// react-spotify-web-playback/esm/components/Loader.js
import * as React from 'react';
import { keyframes, px, styled } from '../styles';
var Wrapper = styled('div')({
    margin: '0 auto',
    position: 'relative',
    '> div': {
        borderRadius: '50%',
        borderStyle: 'solid',
        borderWidth: 0,
        boxSizing: 'border-box',
        height: 0,
        left: '50%',
        position: 'absolute',
        top: '50%',
        transform: 'translate(-50%, -50%)',
        width: 0,
    },
}, function (_a) {
    var styles = _a.styles;
    var pulse = keyframes({
        '0%': {
            height: 0,
            width: 0,
        },
        '30%': {
            borderWidth: px(8),
            height: px(styles.loaderSize),
            opacity: 1,
            width: px(styles.loaderSize),
        },
        '100%': {
            borderWidth: 0,
            height: px(styles.loaderSize),
            opacity: 0,
            width: px(styles.loaderSize),
        },
    });
    return {
        height: px(styles.loaderSize),
        width: px(styles.loaderSize),
        '> div': {
            animation: pulse + " 1.15s infinite cubic-bezier(0.215, 0.61, 0.355, 1)",
            borderColor: styles.loaderColor,
        },
    };
}, 'LoaderRSWP');
function Loader(_a) {
    var styles = _a.styles;
    return (React.createElement(Wrapper, { styles: styles },
        React.createElement("div", null)));
}
export default Loader;
